transform/views/main.py

```diff
@@ -54,236 +47,50 @@ def server_error(error=None):
-            transformer = CSTransformer(logger, survey, survey_response, batch_number, sequence_no)
-
-        transformer.create_zip()
-
-    except Exception as e:
-        tx_id = survey_response.get("tx_id")
-        logger.exception("CS:could not create files for survey", survey_id=survey_id, tx_id=tx_id)
-        return server_error(e)
-
-    logger.info("CS:SUCCESS")
-
-    return send_file(transformer.image_transformer.get_zip(), mimetype='application/zip', add_etags=False)
+def common_software(sequence_no=1000):
+    return transform(sequence_no)
 
 
 @app.route('/cora', methods=['POST'])
 @app.route('/cora/<sequence_no>', methods=['POST'])
 def cora(sequence_no=1000):
-    survey_response = request.get_json(force=True)
-
-    if sequence_no:
-        sequence_no = int(sequence_no)
-
-    survey = get_survey(survey_response)
-    if not survey:
-        return client_error("CORA:Unsupported survey/instrument id")
-
-    survey_id = survey_response.get("survey_id")
-    try:
-        if survey_id == "144":
-            transformer = UKISTransformer(survey_response, sequence_no)
-            transformer.create_zip()
-        else:
-            return client_error("CORA survey with survey id {} is not supported".format(survey_id))
-
-    except Exception as e:
-        tx_id = survey_response.get("tx_id")
-        logger.exception("CORA:could not create files for survey", survey_id=survey_id, tx_id=tx_id)
-        return server_error(e)
-
-    logger.info("CORA:SUCCESS")
-
-    return send_file(transformer.image_transformer.get_zip(), mimetype='application/zip', add_etags=False)
+    return transform(sequence_no)
 
 
 @app.route('/cord', methods=['POST'])
 @app.route('/cord/<sequence_no>', methods=['POST'])
 def cord(sequence_no=1000):
+    return transform(sequence_no)
+
+
+@app.route('/transform', methods=['POST'])
+@app.route('/transform/<sequence_no>', methods=['POST'])
+def transform(sequence_no=1000):
     survey_response = request.get_json(force=True)
 
     if sequence_no:
         sequence_no = int(sequence_no)
 
-    survey = get_survey(survey_response)
-    if not survey:
-        return client_error("CS:Unsupported survey/instrument id")
-
-    survey_id = survey_response.get("survey_id")
     try:
-        if survey_id == "187":
-            if survey_response['collection']['instrument_id'] in ['0001', '0002']:
-                transformer = Ecommerce2019Transformer(survey_response, sequence_no)
-            else:
-                transformer = EcommerceTransformer(survey_response, sequence_no)
-            transformer.create_zip()
-        else:
-            return client_error("CORD survey with survey id {} is not supported".format(survey_id))
+        transformer = get_transformer(survey_response, sequence_no)
+        zip_file = transformer.get_zip()
+        logger.info("TRANSFORM:SUCCESS")
+        return send_file(zip_file, mimetype='application/zip', add_etags=False)
+
+    except MissingIdsException as mie:
```

Comment:
very minor, but having exceptions be `as e:` is widely used around the codebase, can you change it to that?

Reply:
Didn't realise I could have multiple exceptions with the same name - i thought Python was functionally scoped!